biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.ts

```diff
@@ -0,0 +1,44 @@
+import { Town } from '@biogrid/grid-simulator';
+import { Building } from '../building';
+
+/**
+ * A rural area that represents a community where a microgrid can function.
+ */
+export class RuralArea implements Town {
+
+  private town: Building[];
+
+  /**
+   * @param {Building[]} town A list of buildings which make up a town.
+   */
+  constructor(town: Building[]) {
+    this.town = town;
+  }
+
+  getTown() : Building[] {
+    return this.town;
+  }
+
+  /**
+   * This method gets a building in a town by its name identifier.
+   * @param {Building} building The building we're looking for.
+   * @return {Building} Returns the building if found in the list, null if not.
+   */
+  getBuildingByName(building: Building) {
```

Comment:
Given the updated function, your parameter should just be the building name, not the full building info.

Reply:
Thanks will fix that.